components/BottomSheet/examples/BottomSheetTableViewExample.swift

```diff
@@ -0,0 +1,122 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import MaterialComponents.MaterialBottomSheet
+import MaterialComponents.MaterialButtons
+import MaterialComponents.MaterialButtons_ButtonThemer
+
+class BottomSheetTableViewExample: UIViewController {
+  var colorScheme = MDCSemanticColorScheme()
+  var typographyScheme = MDCTypographyScheme()
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+
+    self.title = "Table View Menu"
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    view.backgroundColor = colorScheme.backgroundColor
+
+    let button = MDCButton()
+    button.setTitle("Show bottom sheet", for: .normal)
+    button.addTarget(self,
+                     action: #selector(BottomSheetTableViewExample.didTapFloatingButton),
+                     for: .touchUpInside)
+
+    let buttonScheme = MDCButtonScheme()
+    buttonScheme.colorScheme = colorScheme
+    buttonScheme.typographyScheme = typographyScheme
+    MDCContainedButtonThemer.applyScheme(buttonScheme, to: button)
+
+    button.sizeToFit()
+    button.center = CGPoint(x: view.bounds.midX, y: view.bounds.midY)
```

Comment:
button.center = view.center?

Reply:
The view's center is in its superview's coordinate system. Here it seems the button is being placed within the view's coordinate system.

https://developer.apple.com/documentation/uikit/uiview/1622627-center

Reply:
Assigning a parent’s center to a child’s center is an anti-pattern because it assumes that the parent’s frame origin is 0, 0. I realize we can make that assumption here, but I’d like to encourage a safer coding pattern.

Reply:
it looks like the same here but if its a coding pattern that sounds fair.

Reply:
view.bounds.midX != view.center.x in all cases, most notably if view.frame.origin.x != 0.

Center is in relation to the parent coordinate space as Rob pointed out, not the bounds’.

Reply:
definitely not in all cases, in this case it's the same.